homeassistant/core.py

```diff
@@ -1388,7 +1388,7 @@ def call(
         context: Context | None = None,
         limit: float | None = SERVICE_CALL_LIMIT,
         target: dict | None = None,
-    ) -> bool | None:
+    ) -> dict | None:
```

Comment:
Should we make this a typed dict from the start?

Reply:
Yes, sure.
Should we have any requirements on the result of the service call, for example that it must itself be a JSON serializable dict?

Reply:
Depends on how the result is consumed? That is not clear in this PR at this point. Do we expect custom data in it?

The result can contain things like missing items (which is nested again), I wonder if that is even needed? 

Instead of:
- completed
- result
  - unsupported
  - unavailable
  - ...

It could be just flat
- completed
- unsupported
- unavailable
- ...

Reply:
In the PR, automation traces will include the result dict, but it will simply be showed in the UI.

We probably need to support some level of nesting since a service call can in turn call other services or cascade the call to multiple entities?